ci/do_ci.sh

```diff
@@ -171,6 +171,21 @@ elif [[ "$1" == "bazel.coverage" ]]; then
   # directory. Wow.
   cd "${ENVOY_BUILD_DIR}"
   SRCDIR="${GCOVR_DIR}" "${ENVOY_SRCDIR}"/test/run_envoy_bazel_coverage.sh
+  exit 0
+elif [[ "$1" == "bazel.clang_tidy" ]]; then
+  setup_clang_toolchain
+  echo "Generating compilation database..."
+  cd "${ENVOY_CI_DIR}"
+
+  # The compilation database generate script doesn't support passing build options via CLI.
+  # Writing them into bazelrc
+  echo "build ${BAZEL_BUILD_OPTIONS}" >> tools/bazel.rc
+
+  # bazel build need to be run to setup virtual includes, generating files which are consumed
+  # by clang-tidy
+  "${ENVOY_SRCDIR}/tools/gen_compilation_database.py" --run_bazel_build
+  run-clang-tidy-7
```

Comment:
Sorry where is this defined? Is this the name of the binary that is installed?

Reply:
Yes, if you install `clang-tidy-7` package it comes with a tool to run clang-tidy over compile database:
https://github.com/llvm-mirror/clang-tools-extra/blob/release_70/clang-tidy/tool/run-clang-tidy.py